_source/_posts/2022-06-17-simple-crud-react-and-spring-boot.md

```diff
@@ -0,0 +1,1350 @@
+}
+```
+
+To load some default data, create an `Initializer.java` class in the `com.okta.developer.jugtours` package.
+
+```java
+package com.okta.developer.jugtours;
+
+import com.okta.developer.jugtours.model.Event;
+import com.okta.developer.jugtours.model.Group;
+import com.okta.developer.jugtours.model.GroupRepository;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.stream.Stream;
+
+@Component
+class Initializer implements CommandLineRunner {
+
+    private final GroupRepository repository;
+
+    public Initializer(GroupRepository repository) {
+        this.repository = repository;
+    }
+
+    @Override
+    public void run(String... strings) {
+        Stream.of("Seattle JUG", "Denver JUG", "Dublin JUG",
+                "London JUG").forEach(name ->
+                repository.save(new Group(name))
+        );
+
+        Group djug = repository.findByName("Seattle JUG");
+        Event e = Event.builder().title("Micro Frontends for Java Developers")
+                .description("JHipster now has microfrontend support!")
+                .date(Instant.parse("2022-09-13T17:00:00.000Z"))
+                .build();
+        djug.setEvents(Collections.singleton(e));
+        repository.save(djug);
+
+        repository.findAll().forEach(System.out::println);
+    }
+}
+```
+
+**TIP:** If your IDE has issues with `Event.builder()`, you need to turn on annotation processing and/or install the Lombok plugin. I had to uninstall/reinstall the Lombok plugin in IntelliJ IDEA to get things to work.
+
+If you start your app (using `./mvnw spring-boot:run`) ...
+
+... it will fail because Spring Boot 2.7.0 forces H2 v2.0. The H2 2.0 ecosystem doesn't seem like it's quite ready for prime time, so I recommend you downgrade to H2 version `1.4.200` in your `pom.xml`.
+
+```xml
+<dependency>
+  <groupId>com.h2database</groupId>
+  <artifactId>h2</artifactId>
+  <scope>runtime</scope>
+  <version>1.4.200</version>
+</dependency>
+```
+
+Then, `mvn spring-boot:run` should result in something like:
+
+```
+Group(id=1, name=Seattle JUG, address=null, city=null, stateOrProvince=null, country=null, postalCode=null, user=null, 
+  events=[Event(id=5, date=2022-09-13T17:00:00Z, title=Micro Frontends for Java Developers, description=JHipster now has microfrontend support!, attendees=[])])
+Group(id=2, name=Denver JUG, address=null, city=null, stateOrProvince=null, country=null, postalCode=null, user=null, events=[])
+Group(id=3, name=Dublin JUG, address=null, city=null, stateOrProvince=null, country=null, postalCode=null, user=null, events=[])
+Group(id=4, name=London JUG, address=null, city=null, stateOrProvince=null, country=null, postalCode=null, user=null, events=[])
+```
+
+Add a `GroupController.java` class (in `src/main/java/.../jugtours/web/GroupController.java`) that allows you to CRUD groups.
```

Comment:
"that allows you to CRUD groups." ---> is this as intended? Does CRUD function as the verb, or did a verb go missing?

Reply:
I think it's fine, thanks!